examples/react-fcl-demo/src/demo/DeployContract.tsx

```diff
@@ -0,0 +1,101 @@
+import React, {useState} from "react"
+
+import * as fcl from "@onflow/fcl"
+import * as t from "@onflow/types"
+
+import Card from '../components/Card'
+import Header from '../components/Header'
+import Code from '../components/Code'  
+
+const deployTransaction = `\
+transaction(code: String) {
+  prepare(acct: AuthAccount) {
+    acct.setCode(code.decodeHex())
+  }
+}
+`
+
+const simpleContract = `
+pub contract HelloWorld {
+  pub let greeting: String
+  pub event HelloEvent(message: String)
+
+  init() {
+    self.greeting = "Hello, World!"
+  }
+
+  pub fun hello(message: String): String {
+    emit HelloEvent(message: message)
+    return self.greeting
+  }
+}
+`
+
+const DeployContract = () => {
+  const [status, setStatus] = useState("Not started")
+  const [transaction, setTransaction] = useState(null)
+
+  const runTransaction = async (event: any) => {
+    event.preventDefault()
+    
+    setStatus("Resolving...")
+
+    const blockResponse = await fcl.send([
+      fcl.getLatestBlock(),
+    ])
+
+    const block = await fcl.decode(blockResponse)
+    
+    try {
+      const { transactionId } = await fcl.send([
+        fcl.transaction(deployTransaction),
+        fcl.args([
+          fcl.arg(
+            Buffer.from(simpleContract, "utf8").toString("hex"),
+            t.String
+          )
+        ]),
+        fcl.proposer(fcl.currentUser().authorization),
+        fcl.authorizations([
+          fcl.currentUser().authorization
+        ]),
+        fcl.payer(fcl.currentUser().authorization),
+        fcl.ref(block.id),
```

Comment:
FCL deals with this for you: https://github.com/onflow/flow-js-sdk/blob/master/packages/fcl/src/resolve/index.js#L14

Reply:
fixed